parser.hh

```diff
@@ -199,9 +206,15 @@ class Input
 	 * so that the first request will trigger a fetch from the underlying
 	 * storage.
 	 */
-	Input() : buffer(0), buffer_start(1), buffer_end(0) {}
+	Input(std::string name)
+		: user_name(name), buffer(0), buffer_start(1), buffer_end(0) {}
 	private:
 	/**
+	 * A user-meaningful name.
+	 * This will typically be a filename, but it doesn't have to be.
+	 */
+	std::string	user_name;
```

Comment:
Is there a reason for not making this `const`?  We're not exposing any mechanism for changing it.